### ISSUE TITLE
Document for Slices Pattern

### ISSUE THREAD

**Issue by devanshj** [pmndrs/zustand#1253]
From https://github.com/pmndrs/zustand/pull/1177#issuecomment-1209592975

> I think the problem is that the readme (and some other docs too) link to the typescript document for slices pattern...
> > You may wonder how to organize your code for better maintenance: [Splitting the store into separate slices](https://github.com/pmndrs/zustand/blob/main/docs/typescript.md#slices-pattern)
> 
> Instead there should be a seperate document for slices pattern (just like there used to be a wiki) or a section in readme.

Currently there are a couple of links in some documents that link to the TypeScript guide for slices pattern. Which isn't great because the TypeScript guide only documents the TypeScript instructions and code for slices pattern, it does not introduce or explain slices pattern itself, nor it should, there should be a seperate document for that.

Opening just for tracking purpose, in case someone is interested in working on this.

**Comment by dai-shi** [comment]
cc @chrisk-7777

**Comment by chrisk-7777** [comment]
Thanks @devanshj - just to confirm, we should have:
 - one dedicated page for slices in plain JS
 - a section within the TypeScript page to cover how to apply types to the slices pattern 
 - probably cross-linked to each other (slice page to typescript section, and typescript section to slice page)

**Comment by devanshj** [comment]
Yes, except we already have [a section in TypeScript page regarding slices pattern](https://github.com/pmndrs/zustand/blob/main/docs/guides/typescript.md#slices-pattern) (which is the second point).

**Comment by orangeswim** [comment]
https://github.com/pmndrs/zustand/blob/main/docs/guides/typescript.md#slices-pattern
and https://github.com/pmndrs/zustand/blob/main/docs/guides/flux-inspired-practice.md
have two different ways of doing slices.
The slices pattern is not clear enough in itself on how to easily create a store with distinct slices.
For example, the slices-pattern doesn't explain clearly how I can persist one slice and not persist another. (not specific to persist, but it can by any middleware)

**Comment by devanshj** [comment]
I think with #1344 and #1353 this can be closed?

**Comment by devanshj** [comment]
> https://github.com/pmndrs/zustand/blob/main/docs/guides/typescript.md#slices-pattern
and https://github.com/pmndrs/zustand/blob/main/docs/guides/flux-inspired-practice.md
have two different ways of doing slices.

I think that both are doing the same thing, it's just that the flux version does not extract out the slices. That is this snippet from the flux doc...

```js
const useBoundStore = create((set) => ({
  storeSliceA: ...,
  storeSliceB: ...,
  storeSliceC: ...,
  dispatchX: () => set(...),
  dispatchY: () => set(...),
}))
```

Can be refactored into...

```js
const storeSliceA = (set) => ({ storeSliceA: ..., dispatchX: () => set(...) })
const storeSliceB = (set) => ({ storeSliceB: ..., dispatchY: () => set(...) })
const storeSliceC = (set) => ({ storeSliceC: ... })

const useBoundStore = create((...a) => ({
  ...storeSliceA(...a),
  ...storeSliceB(...a),
  ...storeSliceC(...a),
}))
```

**Comment by dai-shi** [comment]
> Can be refactored into... (Although not suggesting it should be, just pointing out)
> 
> ```js
> const storeSliceA = (set) => ({ storeSliceA: ..., dispatchX: () => set(...) })
> const storeSliceB = (set) => ({ storeSliceB: ..., dispatchY: () => set(...) })
> const storeSliceC = (set) => ({ storeSliceC: ... })
> 
> const useBoundStore = create((...a) => ({
>   ...storeSliceA(...a),
>   ...storeSliceB(...a),
>   ...storeSliceC(...a),
> }))
> ```

No, that is not a Flux pattern as I understand. Actions should be isolated from slices. That's my intention in the original snippet.

**Comment by devanshj** [comment]
I see, but we're anyway already far away from the "real flux" :P

**Comment by dai-shi** [comment]
😄 So, I'd say, flux pattern and slice pattern are different thing.

**Comment by devanshj** [comment]
Maybe then we should remove this link from the flux guide then?

> See [Splitting the store into separate slices](https://github.com/pmndrs/zustand/blob/main/docs/guides/typescript.md#slices-pattern) for how to define a store with separate slices.

This makes it sound like slices pattern can be used with flux pattern.

Or maybe I'm misreading and the link is there to mention slices pattern as an alternative to flux pattern? If that's the case then perhaps it should be worded more clearly.

**Comment by dai-shi** [comment]
Yeah, I don't mind if my original intention is kept somewhere.
Let's have @chrisk-7777 in loop.

**Comment by devanshj** [comment]
Closing as the issue in OP is resolved with #1344 and #1353.
